sdk/modelsrepository/azure-iot-modelsrepository/src/test/java/com/azure/iot/modelsrepository/ModelRepositoryIntegrationTests.java

```diff
@@ -42,6 +42,31 @@ public void getModelsSingleDtmiWithDeps(HttpClient httpClient, ModelsRepositoryS
 
         Assertions.assertTrue(result.keySet().stream().count() == expectedDtmis.stream().count());
         Assertions.assertTrue(result.keySet().containsAll(expectedDependencies));
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.iot.modelsrepository.TestHelper#getTestParameters")
+    public void getModelsEnsureNoDuplicates(HttpClient httpClient, ModelsRepositoryServiceVersion serviceVersion) throws URISyntaxException {
+        List<String> inputDtmis = Arrays.asList(
+            "dtmi:azure:DeviceManagement:DeviceInformation;1",
+            "dtmi:azure:DeviceManagement:DeviceInformation;1"
+        );
+
+        ModelsRepositoryAsyncClient client = getAsyncClient(httpClient, serviceVersion);
+
+        Map<String, String> result = client.getModels(inputDtmis).block();
+        Assertions.assertTrue(result.keySet().stream().count() == 1);
+    }
 
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.iot.modelsrepository.TestHelper#getTestParameters")
+    public void getModelsSingleDtmiWithDepsDisableDependencyResolution(HttpClient httpClient, ModelsRepositoryServiceVersion serviceVersion) throws URISyntaxException {
+        final String dtmi = "dtmi:com:example:Thermostat;1";
+        ModelsRepositoryAsyncClient client = getAsyncClient(httpClient, serviceVersion);
```

Comment:
Why get the async client, but then block on it? We could test the sync client (instead?) or call it in an async way as we'd expect a customer to, right?

Reply:
absolutely, I am just testing the logic for now to make sure I am on the right track, these will all change shortly when the PR is ready for review.